### DISCUSSION TITLE
Instructions on how to deploy production apps to a remote Ubuntu server?

### DISCUSSION THREAD

**Discussion by Lepiloff** [fastapi/fastapi#7343] [Questions]
Hi!
I am looking for a way to deploy my application (Fastapi, Postgresql) from the local machine to a remote Linux server (digitalocean). All I find on the network is a sample deploying on heroku with a mysqlite database. Is there a complete deploying guide with a detailed step-by-step description? The web is full of similar articles on deploying flask or django and there is nothing about the fastapi framework.

**Comment by ArcLightSlavik** [comment]
Can try this:

https://towardsdatascience.com/deploy-python-tesseract-opencv-fast-api-using-aws-ec2-instance-ed3c0e3f2888
https://towardsdatascience.com/how-to-deploy-your-machine-learning-web-app-to-digital-ocean-64bd19ce15e2

**Comment by GabrielCarpr** [comment]
If you've got any experience with Docker (if you don't, I'd suggest learning. It makes deploying apps very easy), then you could use the FastAPI docker image. It's all set up ready to use, so all you'd have to do it start it up, set the port and configure SSL, and give your app the database credentials.

Most tutorials mentioning SQLite are running the database on the production server. Instead with postgres you could use a DigitalOcean managed database, which will probably be easiest for you.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:

**Comment by windson** [comment]
[This ](https://bit.ly/3hpS0Ws) video tutorial should help you deploying FastAPI with PostgreSQL, but it does on Ubuntu VM created on Azure.
